interdiff.go

```diff
@@ -87,40 +84,222 @@ func compareFileDiff(oldFileDiff, newFileDiff *diff.FileDiff) *diff.FileDiff {
+
+	// Compare, while there are hunks to process
+	for (currentOldHunkI < len(oldHunks)) || (currentNewHunkJ < len(newHunks)) {
+
+		// Entering next hunk in oldHunks
+		if (currentOldHunkI < len(oldHunks)) && (i == -1) && (currentOrgI == oldHunks[currentOldHunkI].OrigStartLine) {
+			i = 0
+			oldHunkBody = strings.Split(string(oldHunks[currentOldHunkI].Body), "\n")
+			// Remove empty line in the end
+			if oldHunkBody[len(oldHunkBody)-1] == "" {
+				oldHunkBody = oldHunkBody[:len(oldHunkBody)-1]
+			}
+		}
+
+		// Entering next hunk in newHunks
+		if (currentNewHunkJ < len(newHunks)) && (j == -1) && (currentOrgI == newHunks[currentNewHunkJ].OrigStartLine) {
+			j = 0
+			newHunkBody = strings.Split(string(newHunks[currentNewHunkJ].Body), "\n")
+			if newHunkBody[len(newHunkBody)-1] == "" {
+				newHunkBody = newHunkBody[:len(newHunkBody)-1]
+			}
+		}
+
+		switch {
+		case (i == -1) && (j == -1):
+			break
+		case (i >= 0) && (j == -1):
+			newBody = append(newBody, revertedLine(oldHunkBody[i]))
+			// Added one of lines from origin
+			if !strings.HasPrefix(oldHunkBody[i], "+") {
+				currentOrgI++
+			}
+			i++
+
+		case (i == -1) && (j >= 0):
+			newBody = append(newBody, newHunkBody[j])
+			// Added one of lines from origin
+			if !strings.HasPrefix(newHunkBody[j], "+") {
+				currentOrgI++
+			}
+			j++
+
+		default:
+			switch {
+			// Firstly proceeding added lines
+			case strings.HasPrefix(oldHunkBody[i], "+"):
+				newBody = append(newBody, revertedLine(oldHunkBody[i]))
+				i++
+			case strings.HasPrefix(newHunkBody[j], "+"):
+				newBody = append(newBody, newHunkBody[j])
+				j++
+			default:
+				switch {
+				case strings.HasPrefix(oldHunkBody[i], " ") && strings.HasPrefix(newHunkBody[j], " "):
+					newBody = append(newBody, oldHunkBody[i])
+				case strings.HasPrefix(oldHunkBody[i], "-") && strings.HasPrefix(newHunkBody[j], " "):
+					newBody = append(newBody, revertedLine(oldHunkBody[i]))
+				case strings.HasPrefix(oldHunkBody[i], " ") && strings.HasPrefix(newHunkBody[j], "-"):
+					newBody = append(newBody, newHunkBody[j])
+					// If both have deleted same line, no need to append it to newBody
+				}
+
+				currentOrgI++
+				i++
+				j++
+			}
+		}
+
+		if i >= len(oldHunkBody) {
+			i = -1
+			currentOldHunkI++
+		}
+
+		if j >= len(newHunkBody) {
+			j = -1
+			currentNewHunkJ++
+		}
+	}
+
+	resultHunk.Body = []byte(strings.Join(newBody, "\n"))
+
+	for _, line := range newBody {
+		if !strings.HasPrefix(line, " ") {
+			return resultHunk
+		}
 	}
+
+	return nil
+}
+
+func configureResultHunk(oldHunks, newHunks []*diff.Hunk) (*diff.Hunk, int32) {
+	if (len(oldHunks) == 0) || (len(newHunks) == 0) {
+		return nil, 0
+	}
+
+	var currentOrgI int32
+	resultHunk := &diff.Hunk{OrigStartLine: -1,
+		OrigLines:       -1,
+		OrigNoNewlineAt: -1,
+		NewStartLine:    -1,
+		NewLines:        -1,
+		// TODO: Concatenate sections
+		Section:       "",
+		StartPosition: -1,
+		Body:          []byte{0},
+	}
```

Comment:
```suggestion
	resultHunk := &diff.Hunk{
		// TODO: Concatenate sections
		Section:       "",
		Body:          []byte{0},
	}
```

Members that you set below don't need to be initialized.